activerecord/test/cases/associations/has_many_through_associations_test.rb

```diff
@@ -1245,6 +1245,12 @@ def test_has_many_through_with_scope_should_respect_table_alias
     assert_equal 0, users[2].family_members.to_a.size
   end
 
+  def test_has_many_through_with_scope_should_works_properly
```

Comment:
There's an extra "s" in "work" here. Also maybe the method name should mention `preload` as the issue doesn't happen without it, what do you think ?